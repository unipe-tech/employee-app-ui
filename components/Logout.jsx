import { Pressable, Text } from "react-native";
import { AntDesign } from "react-native-vector-icons";
import { useDispatch } from "react-redux";
import { useNavigation } from "@react-navigation/core";

import { resetAadhaar } from "../store/slices/aadhaarSlice";
import { resetAuth } from "../store/slices/authSlice";
import { resetBank } from "../store/slices/bankSlice";
import { resetESIC } from "../store/slices/esicSlice";
import { resetLicense } from "../store/slices/licenseSlice";
import { resetNavigation } from "../store/slices/navigationSlice";
import { resetPan } from "../store/slices/panSlice";
import { resetProfile } from "../store/slices/profileSlice";
import { resetTimer } from "../store/slices/timerSlice";

const Logout = () => {

  const dispatch = useDispatch();
  const navigation = useNavigation();

  // const store = useSelector((state) => state);
  // console.log(store);
  return (
    <Pressable
      onPress={() => {
        dispatch(resetAuth());
        dispatch(resetNavigation());
        dispatch(resetProfile());
        dispatch(resetPan());
        dispatch(resetAadhaar());
        dispatch(resetESIC());
        dispatch(resetBank());
        dispatch(resetLicense());
        dispatch(resetTimer());
        
        navigation.navigate("Login");
      }}
      style={{
        flexDirection: "row",
        alignItems: "center",
        paddingHorizontal: 20,
        paddingVertical: 20,
      }}
    >
      <AntDesign name="logout" color="black" size={16} />
      <Text style={{ fontSize: 16, paddingLeft: 10 }}>Logout</Text>
    </Pressable>
  );
};

export default Logout;