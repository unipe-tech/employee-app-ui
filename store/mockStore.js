const store = {
  // Auth

  auth: {
    id: 1,
    phoneNumber: "9876543210",
    verifyStatus: "PENDING",
  },

  // Aadhaar

  aadhaar: {
    data: {
      aadhaar_data: {
        name: "Xyz garg",
      },
    },
    number: "789456123010",
    submitOTPtxnId: "987",
    verifyMsg: "This otp has been sent",
    verifyStatus: "PENDING",
  },

  // Bank

  bank: {
    accountNumber: "157894561230",
    ifsc: "INDB0000123",
    upi: "9876543210@upi",
    accountHolderName: "Xyz Goyal",
    verifyStatus: "PENDING",
    verifyMsg: "The verification is pending",
    bankName: "IndusInd Bank",
    bankBranch: "Mumbai",
    branchCity: "Mumbai",
  },

  // ESIC

  esic: {
    portal: { estCode: "", ipNumber: "" },
    familyDetails: {
      fatherHusband: { relation: "", name: "" },
      nominee: { relation: "", name: "" },
    },
    address: {
      present: { street: "", state: "", district: "", pincode: "" },
      permanent: { street: "", state: "", district: "", pincode: "" },
      nominee: { street: "", state: "", district: "", pincode: "" },
    },
  },

  // License

  license: {
    number: "",
    name: "",
    dob: "",
    bloodGroup: "",
    classes: {},
    rto: {},
    validity: {},
    verifyMsg: "",
    verifyStatus: "PENDING",
  },

  // Navigation

  navigation: { currentScreen: "Welcome" },

  // PAN

  pan: {
    dob: "02/02/2002",
    email: "xyz@unipe.co",
    gender: "Male",
    name: "Xyz Goyal",
    number: "ABCDE1234F",
    verifyMsg: "Verification Pending",
    verifyStatus: "PENDING",
  },

  // Profile

  profile: {
    maritalStatus: "Married",
    educationalQualification: "12th pass",
    alternatePhone: "+913216549870",
    email: "abc@unipe.co",
    selfie: "",
  },

  // Timer

  timer: { login: 2 * 60, aadhaar: 10 * 60 },
};

export default store;
