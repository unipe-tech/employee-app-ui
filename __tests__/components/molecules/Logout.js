import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { describe, expect, test } from "@jest/globals";
import Logout from "../../../components/molecules/Logout";
import configureStore from "redux-mock-store";
import mockStore from "../../../store/mockStore";
import { NavigationContainer } from "@react-navigation/native";
import { Provider, useSelector } from "react-redux";

const middleware = [];

const mockedStore = configureStore(middleware);

const createdStore = mockedStore(mockStore);

describe("Logout", () => {
  test("renders UI correctly", () => {
    const tree = render(
      <Provider store={createdStore}>
        <NavigationContainer>
          <Logout />
        </NavigationContainer>
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
