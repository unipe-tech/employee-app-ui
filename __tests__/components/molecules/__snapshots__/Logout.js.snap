// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout renders UI correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingHorizontal": 20,
      "paddingVertical": 20,
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "color": "#f56a6a",
        "fontFamily": "Montserrat-Bold",
        "fontSize": 14,
        "lineHeight": 18,
        "paddingLeft": 10,
      }
    }
  >
    Logout
  </Text>
</View>
`;
